game-core/src/main/java/games/strategy/triplea/delegate/BattleDelegate.java

```diff
@@ -819,28 +814,26 @@ private void doScrambling() {
         continue;
       }
       boolean scrambledHere = false;
-      for (final var scramblers : scramblersByTerritoryPlayer.get(terrPlayer)) {
```

Comment:
This loop is no longer needed now that we removed the extra Collection that was previously scramblersByTerritoryPlayer.get(terrPlayer).

We still need the inner loop - it was previous a while() loop on an Iterator, but I switched it to a simpler for loop below on line 821.

Reply:
The inner loop is now a removeIf() instead.